src/components/Card.js

```diff
@@ -1,20 +1,31 @@
 class Card {
-  constructor({ name, link }, cardSelector, cardImageClickHandler) {
+  constructor(
+    { name, link, _id, isLiked },
+    cardSelector,
+    cardImageClickHandler,
+    handleDelete,
+    cardLikeClick
+  ) {
+    this._isLiked = isLiked;
+    this._handleCardLikeClick = cardLikeClick;
     this._name = name;
     this._link = link;
+    this.id = _id;
     this._cardSelector = cardSelector;
     this._cardImageClickHandler = cardImageClickHandler;
+    this._handleDelete = handleDelete;
   }
 
   _setEventListeners() {
-    const likeButton = this._cardElement.querySelector(".card__like-button");
-    likeButton.addEventListener("click", () => {
-      likeButton.classList.toggle("card__like-button_active");
+    this._likeButton.addEventListener("click", () => {
+      this._isLiked = !this._isLiked;
+      this.updateLikeState();
```

Comment:
You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.